draft_python/mtg_draft_ai/controller.py

```diff
@@ -0,0 +1,128 @@
+import random
+from mtg_draft_ai.api import *
+
+
+class DraftController:
+    """Runs a draft by asking each drafter to pick from the right pack in the right order."""
+
+    def __init__(self, draft_info, drafters, packs):
+        """Basic init method which does no manipulation of its parameters.
+
+        For most cases, use DraftController.create instead, which creates shuffled packs
+        from the card list.
+
+        Args:
+            draft_info (api.DraftInfo): Config about the draft, which will not change
+                during the draft.
+            drafters (list of api.Drafter): The Drafters, which may have different Picker
+                implementations.
+            packs (api.Packs): The Packs to use for this draft. Should already be initialized.
+        """
+        self.draft_info = draft_info
+        self.drafters = drafters
+        self.packs = packs
```

Comment:
Should we validate the length of `drafters` as we do below? Similarly that the packs have the right lengths?

Reply:
I guess I'm not excited to add a lot of validation here given that there are a lot of potential checks required to "guarantee" that everything is consistent.

The approach I'm aiming for is to have the `create` method be the main way to create instances, which has validation logic and creates valid packs. Given that, it doesn't feel like there's a lot of bang for buck to adding the validation here too.